pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -3259,17 +3260,17 @@ func (c *Cloud) getInstancesByNodeNamesCached(nodeNames sets.String) ([]*ec2.Ins
 	return instances, nil
 }
 
-func (c *Cloud) getInstancesByNodeNames(nodeNames []string) ([]*ec2.Instance, error) {
+func (c *Cloud) getInstancesByNodeNames(nodeNames []string, states ...string) ([]*ec2.Instance, error) {
 	names := aws.StringSlice(nodeNames)
```

Comment:
I only looked at gce but, since gce does not filter by "state" in their equivalent-ish `getInstancesByNames` IMO it's plain wrong for `getinstancesByNodeNames` to default to only getting "running" instances. I think it's better if we change all the other call sites and force everyone in the future using this function to decide if they want to filter node instances by states